hibernate-core/src/main/java/org/hibernate/query/sql/internal/ParameterParser.java

```diff
@@ -178,6 +176,14 @@ else if ( c == '?' ) {
 		recognizer.complete();
 	}
 
+	private static String preprocessing(String sqlString) {
+		final Map<String, String> preprocessingExchangeMap = Map.of("::=", ":=");
+		for(Map.Entry<String, String> entry: preprocessingExchangeMap.entrySet()) {
+			sqlString = sqlString.replace(entry.getKey(), entry.getValue());
```

Comment:
IMO we should also warn people about the use of deprecated escaping.

```suggestion
			final String newSqlString = sqlString.replace(entry.getKey(), entry.getValue());
			if ( newSqlString != sqlString ) {
				// DeprecationLogger warning
				sqlString = newSqlString;
			}
```

Reply:
That seems like a very good suggestion. By throwing in a message, we can naturally reduce the instances of incorrect usage. I have added the phrase for this purpose, so please review it.